src/Symfony/Component/ExpressionLanguage/Node/ArrayNode.php

```diff
@@ -58,6 +58,19 @@ public function evaluate($functions, $values)
         return $result;
     }
 
+    public function dump()
+    {
+        $str = '[';
+
+        foreach ($this->getKeyValuePairs() as $pair) {
+            $str .= sprintf('%s => %s, ', $pair['key']->attributes['value'], $pair['value']->dump());
```

Comment:
This dumping logic means that it won't be a valid expression when dumping a hash. Is it expected ? all other dumpers looks like the would create a valid expression again